### PR TITLE
rename CSF 2017 scripts

### PR DESCRIPTION
**Pull request by davidsbailey**
### Background

Implements step 7 in the [script rename dev doc](https://docs.google.com/document/d/1_6-d_Fj4B6T6FYh2ih8g0x2FEwAsmuXLxuYHGhltCX8/edit#heading=h.n4l9d83x5ybz). A similar rename was previously completed for CSP and CSD units in step 5 (https://github.com/code-dot-org/code-dot-org/pull/22839).

### Description
* renames CSF scripts Course A - Course F, Express, and Pre-Express, causing `scripts` table entries to be renamed. A caveats, verification, and a comparison with an alternate approach is detailed in the comments below.

### CODE REVIEW

**Comment by joshlory** [comment]
We're updating a lot of identifiers here that we never expected to change.  That makes me a little nervous.

Have you thought through the option of adding the year to the script route, rather than renaming all the scripts?

**Comment by davidsbailey** [comment]
Good question Josh. I'm out tomorrow so let's discuss on Friday.

**Comment by davidsbailey** [comment]
Hi Josh, thanks for calling attention to this. The two approaches as I understand them are:

1. change url paths by renaming scripts in the DB (proposed here)
    - a. add new_name field to all scripts and update serialize / seed logic (previously done)
    - b. some updates to serving logic (previously done in #22770 )
    - c. review and consider renaming all occurrences of "coursea", .. "coursef", "express", "pre-express" (relatively easy, done in this PR)
    - d. review and consider updating occurrences of the following in dashboard/app (# occurrences): `Script.find_by(` (2), `Script.find_by_name` (7), `Script.get_from_cache` (100+) or `Script.get_without_cache` (3) -- (hard, not really done very thoroughly in this PR)
    - e. other things to check?

2. change script url paths without renaming scripts (proposed alternative)
    - a. add a `url_token` or field on each script (relatively easy)
    - b. update Script.get_without_cache() to look for `Script.find_by(url_token: ...)` instead of `Script.find_by(name: ...)`(relatively easy)
    - c. review other references to script.name and evaluate whether they should be updated to script.url_path (seems hard?)
    - d. review and consider updating all occurrences of `Script.find_by(:name`, `Script.find_by_name`, `Script.get_from_cache` or `Script.get_without_cache` (seems hard, like above)
    - e. other things to check?

In both cases, updating scripts / serialize / seed (step a) and updating the script serving logic (step b) seem straight forward. however, step (2c) seems hard to me whereas I believe I've done a reasonably good job of (1c) in this PR. 

step (d) looks hard to me in both cases and I do NOT believe I've done a great job of it here. the main difference in my mind is that with approach (1), a call to one of the 100+ instances of Script.get_from_cache is very unlikely to break unless its argument is ultimately a hard-coded reference to a CSF course name, which I've searched for as thoroughly as I can. If you can think of another "attack vector" where a Script.get_from_cache could break with approach (1), I'm all ears. by contrast, with approach (2) it seems like i have to go and figure out if it is ultimately pulling a script.name from somewhere and whether that needs to be updated to script.url_path, which seems harder to validate without tracing the codepath for every single call.

In terms of empirical evidence, we used approach (1) above when renaming CSP and CSD units e.g. csp1 --> csp1-2018, and we did have two transient outages on non-critical codepaths during the deploy, one between rake seed and the dashboard update, and one between the dashboard update and the javascript update. This result is neither particularly bad nor particularly good -- it certainly leaves the opening for a more severe breakage if CSF scripts are updated in the same way.

So that's my best understanding of the trade-offs. please let me know what you think, especially on how to proceed down either path with higher confidence in not breaking things too badly.

CC @islemaster and @balderdash who have also reviewed other PRs in this feature area recently.

more details about what has been proposed / done are in the [script rename dev doc](https://docs.google.com/document/d/1_6-d_Fj4B6T6FYh2ih8g0x2FEwAsmuXLxuYHGhltCX8/edit#), though alternate proposals were not discussed there.

**Comment by islemaster** [comment]
I thought a major advantage of renaming the scripts was that we can pretty much use existing levelbuilder tools without modification (they're just scripts with different names), where adding the year to the script route would probably require more work extending our internal tools to understand the concept of versioned scripts vs non-versioned scripts.

File: dashboard/test/models/script_test.rb [hunk 1]
```diff
@@ -541,21 +541,21 @@ def populate_cache_and_disconnect_db
   end
 
   test 'summarize includes show_script_version_warning' do
-    a17 = create(:script, name: 'coursea-2017', family_name: 'coursea')
-    a18 = create(:script, name: 'coursea-2018', family_name: 'coursea')
+    foo17 = create(:script, name: 'foo-2017', family_name: 'foo')
+    foo18 = create(:script, name: 'foo-2018', family_name: 'foo')
```
Comment (islemaster) [inline dashboard/test/models/script_test.rb:545]: :+1: Good call switching to fake script names in unit tests.

**Comment by davidsbailey** [comment]
Thanks for the review, @islemaster ! If there are no comments from other reviewers, I'll merge this once it passes circle again.

CC @sureshc for contact rollups, addressed in this PR
CC @bencodeorg for aws/redshift, not covered by this PR

**Comment by bencodeorg** [comment]
Sounds good @davidsbailey, I'll handle the redshift type fallout once the renames happen.

**Comment by davidsbailey** [comment]
heads up @wjordan (DotD) that there could be some transient hb errors related to scripts `coursea` ... `coursef`, `express`, and `preexpress` during Tuesday's deploy. feel free to ping me if this happens.

**Pull request merged**